packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -1,7 +1,72 @@
 import numpy as np
-from matplotlib.backend_bases import GraphicsContextBase, RendererBase
+from matplotlib.backends.browser_backend import \
+    FigureCanvasWasm, NavigationToolbar2Wasm
+from matplotlib.backend_bases import (
+    GraphicsContextBase, RendererBase,
+    FigureManagerBase, _Backend
+)
+
+from matplotlib.colors import colorConverter, rgb2hex
+from matplotlib.transforms import Affine2D
+from matplotlib.path import Path
+from matplotlib import interactive
+
+from js import document
+
+import base64
+import io
+
+_capstyle_d = {'projecting': 'square', 'butt': 'butt', 'round': 'round'}
+
+interactive(True)
+
+
+class FigureCanvasHTMLCanvas(FigureCanvasWasm):
+
+    def __init__(self, *args, **kwargs):
+        FigureCanvasWasm.__init__(self, *args, **kwargs)
+
+    def draw(self):
+        # Render the figure using custom renderer
+        self._idle_scheduled = True
+        orig_dpi = self.figure.dpi
+        if self._ratio != 1:
+            self.figure.dpi *= self._ratio
+        try:
+            width, height = self.get_width_height()
+            canvas = self.get_element('canvas')
+            if canvas is None:
+                return
+            ctx = canvas.getContext("2d")
+            renderer = RendererHTMLCanvas(ctx, width, height, dpi=72)
```

Comment:
The DPI is hard-coded to 72 here, so it won't take into account the resolution of the display.  I think this might be the reason we're seeing thin lines here.  Try passing in `self.figure.dpi`.